commands/command_test.go

```diff
@@ -6,7 +6,199 @@ import (
 	"github.com/spf13/cobra"
 )
 
-func (s *MmctlUnitTestSuite) TestCommandListCmdF() {
+func (s *MmctlUnitTestSuite) TestCommandCmdF() {
+	s.Run("Create a new custom slash command for a specified team", func() {
+		printer.Clean()
+		teamArg := "example-team-id"
+		titleArg := "example-command-name"
+		descriptionArg := "example-description-text"
+		triggerWordArg := "example-trigger-word"
+		urlArg := "http://localhost:8000/example"
+		creatorIdArg := "example-user-id"
+		creatorUsernameArg := "example-user"
+		responseUsernameArg := "example-username2"
+		iconArg := "icon-url"
+		method := "G"
+		autocomplete := false
+		autocompleteDesc := "autocompleteDesc"
+		autocompleteHint := "autocompleteHint"
+
+		mockTeam := model.Team{Id: teamArg}
+		mockUser := model.User{Id: creatorIdArg, Username: creatorUsernameArg}
+		mockCommand := model.Command{
+			TeamId:           teamArg,
+			DisplayName:      titleArg,
+			Description:      descriptionArg,
+			Trigger:          triggerWordArg,
+			URL:              urlArg,
+			CreatorId:        creatorIdArg,
+			Username:         responseUsernameArg,
+			IconURL:          iconArg,
+			Method:           method,
+			AutoComplete:     autocomplete,
+			AutoCompleteDesc: autocompleteDesc,
+			AutoCompleteHint: autocompleteHint,
+		}
+
+		cmd := &cobra.Command{}
+		cmd.Flags().String("team", teamArg, "")
+		cmd.Flags().String("title", titleArg, "")
+		cmd.Flags().String("description", descriptionArg, "")
+		cmd.Flags().String("trigger-word", triggerWordArg, "")
+		cmd.Flags().String("url", urlArg, "")
+		cmd.Flags().String("creator", creatorIdArg, "")
+		cmd.Flags().String("response-username", responseUsernameArg, "")
+		cmd.Flags().String("icon", iconArg, "")
+		cmd.Flags().String("method", method, "")
+		cmd.Flags().Bool("autocomplete", autocomplete, "")
+		cmd.Flags().String("autocompleteDesc", autocompleteDesc, "")
+		cmd.Flags().String("autocompleteHint", autocompleteHint, "")
+
+		// createCommandCmdF will call getTeamFromTeamArg,  getUserFromUserArg which then calls GetUserByEmail
+		s.client.
+			EXPECT().
+			GetTeam(teamArg, "").
+			Return(&mockTeam, &model.Response{Error: nil}).
+			Times(1)
+		s.client.
+			EXPECT().
+			GetUserByEmail(creatorIdArg, "").
+			Return(&mockUser, &model.Response{Error: nil}).
+			Times(1)
+		s.client.
+			EXPECT().
+			CreateCommand(&mockCommand).	
+			Return(&mockCommand, &model.Response{Error: nil}).
+			Times(1)
+
+		err := createCommandCmdF(s.client, cmd, []string{teamArg})
+		s.Require().Nil(err)
+		s.Len(printer.GetLines(), 1)
```

Comment:
Here, after checking the amount of lines printed, we should check as well for the contents of the line:

```go
s.Equal(&mockCommand, getLines()[0})
```

Reply:
done!